module/applications/compendium-browser.mjs

```diff
@@ -1017,6 +1070,14 @@ export default class CompendiumBrowser extends foundry.applications.api.Handleba
         // Apply module art to the new index
         .then(index => game.dnd5e.moduleArt.apply(index)))
 
+        // Derive source values
+        .map(i => {
+          if ( !index ) return i;
```

Comment:
Because we do filtering on the index before fetching the full documents, this will prevent source filter from being used when fetching full documents.

```suggestion
```